### PR TITLE
[MM-30991] Allow webhook commands to specify an owner

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
The owner flag should allow to set an owner different from the `user` and from the requester.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30991

### CODE REVIEW

**Review by lindy65 — APPROVED** [review]
Thanks @mgdelacroix - this will be tested after merging 👍

File: commands/webhook.go [hunk 1]
```diff
@@ -155,6 +212,19 @@ func createIncomingWebhookCmdF(c client.Client, command *cobra.Command, args []s
...
+	var owner *model.User
+	ownerArg, _ := command.Flags().GetString("owner")
+	if viper.GetBool("local") && ownerArg == "" {
+		return errors.New("owner flag should be set when this command is run in local mode")
+	}
+
+	if ownerArg != "" {
+		owner = getUserFromUserArg(c, ownerArg)
+		if owner == nil {
+			return errors.New("unable to find owner user: " + ownerArg)
```
Comment (ethervoid) [inline commands/webhook.go:224]:
```suggestion
			return errors.Errorf("unable to find owner user: ", ownerArg)
```

**Review by ethervoid — APPROVED** [review]
LGTM! just two questions

File: client/client.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ type Client interface {
 	SearchTeams(search *model.TeamSearch) ([]*model.Team, *model.Response)
 	GetPost(postId string, etag string) (*model.Post, *model.Response)
 	CreatePost(post *model.Post) (*model.Post, *model.Response)
-	GetPostsForChannel(channelId string, page, perPage int, etag string) (*model.PostList, *model.Response)
+	GetPostsForChannel(channelId string, page, perPage int, etag string, collapsedThreads bool) (*model.PostList, *model.Response)
```
[the remark this answers is not in the document]
Comment (mgdelacroix) [reply client/client.go:44]: We should, but this is just picking up with what has already been done in `master`.

File: commands/webhook.go (hunk 1 above)
Comment (mgdelacroix) [reply commands/webhook.go]: Addressed in 49bf106

Comment (ethervoid) [reply commands/webhook.go]: https://media.giphy.com/media/4xpB3eE00FfBm/giphy.gif

File: commands/auth.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
-	"golang.org/x/crypto/ssh/terminal"
+	"golang.org/x/term"
```
Comment (isacikgoz) [inline commands/auth.go:17]: I see you are a man of culture.

File: commands/webhook.go [hunk 4]
```diff
@@ -146,13 +203,26 @@ func createIncomingWebhookCmdF(c client.Client, command *cobra.Command, args []s
...
 	user := getUserFromUserArg(c, userArg)
 	if user == nil {
-		return errors.New("Unable to find user '" + userArg + "'")
+		return errors.Errorf("Unable to find user '%s'", userArg)
+	}
+
+	var owner *model.User
+	ownerArg, _ := command.Flags().GetString("owner")
+	if viper.GetBool("local") && ownerArg == "" {
+		return errors.New("owner flag should be set when this command is run in local mode")
```
Comment (isacikgoz) [inline commands/webhook.go:218]:
```suggestion
		return errors.New("owner should be specified to run this command in local mode")
```

same suggestion for other occurrences.

File: commands/webhook_e2e_test.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestCreateIncomingWebhookCmd() {
```
Comment (isacikgoz) [inline commands/webhook_e2e_test.go:15]: Would adding a test case for regular user make sense?

**Review by isacikgoz — COMMENTED** [review]
Awesome improvement! 🚀 A couple of suggestions and questions 👇

File: commands/webhook_e2e_test.go (hunk 5 above)
Comment (mgdelacroix) [reply commands/webhook_e2e_test.go:15]: It totally would :) Addressed in dfb3338

File: commands/webhook.go (hunk 4 above)
Comment (mgdelacroix) [reply commands/webhook.go]: Addressed in dfb3338

**Comment by mgdelacroix** [comment]
@isacikgoz PTAL :smile:

**Review by isacikgoz — APPROVED** [review]
🚀 🚀 🚀

**Comment by hanzei** [comment]
Thank you for your contribution. :+1:  `mmctl` has been moved into https://github.com/mattermost/mattermost/tree/master/server/cmd/mmctl.  Please resubmit your PR to https://github.com/mattermost/mattermost.

**Pull request closed without merging**